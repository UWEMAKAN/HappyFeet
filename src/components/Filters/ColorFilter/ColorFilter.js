import React from 'react';
import propTypes from 'prop-types';
import ColorItem from './Color/Color';
import './ColorFilter.css';

const ColorFilter = ({ checkedHandler }) => {
  const colors = [
    { name: 'Blue', color: '#2c9cee' },
    { name: 'Red', color: '#e00b0b' },
    { name: 'black', color: '#000000' },
    { name: 'White', color: '#ffffff' },
    { name: 'Brown', color: '#8B4513' },
    { name: 'Green', color: '#1e5601' }
  ];

  const items = colors.map((c) => (
    <ColorItem
      checkedHandler={checkedHandler}
      key={c.name}
      color={c.color}
      name={c.name}
    />
  ));
  return (
    <div className="Filter ColorFilter">
      <h3>Color</h3>
      {items}
    </div>
  );
};

ColorFilter.propTypes = {
  checkedHandler: propTypes.func.isRequired
};

export default ColorFilter;
