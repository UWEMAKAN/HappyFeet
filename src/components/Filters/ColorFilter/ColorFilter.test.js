import React from 'react';
import { shallow } from 'enzyme';
import ColorFilter from './ColorFilter';

const renderColorFilter = (args) => {
  const defaultProps = {
    checkedHandler: jest.fn()
  };
  const props = { ...defaultProps, ...args };
  return shallow(<ColorFilter checkedHandler={props.checkedHandler} />);
};

describe('ColorFilter Component', () => {
  test('should render ColorFilter Component', () => {
    const checkedHandler = jest.fn();
    const wrapper = renderColorFilter({ checkedHandler });
    expect.assertions(3);
    expect(wrapper.find('h3').text()).toEqual('Color');
    expect(wrapper.getElements()[0].props.children[1].length).toBe(6);
    expect(wrapper).toMatchSnapshot();
  });
});
