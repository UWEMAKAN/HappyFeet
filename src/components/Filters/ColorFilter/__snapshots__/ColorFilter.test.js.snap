// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorFilter Component should render ColorFilter Component 1`] = `ShallowWrapper {}`;
